src/overlays/actors/ovl_En_fHG/z_en_fhg.c

```diff
@@ -27,29 +61,643 @@ const ActorInit En_fHG_InitVars = {
+        osSyncPrintf("SPD Z %f\n", this->inPaintingVelZ);
+        osSyncPrintf("X=%f\n", sp54);
+        osSyncPrintf("Z=%f\n", sp50);
+        if (sp4C == 0.0f) {
+            this->timers[0] = 140;
+            this->actionFunc = EnfHG_Retreat;
+            Animation_MorphToLoop(&this->skin.skelAnime, &gPhantomHorseAnim_00B4C8, 0.0f);
+            this->bossFhgSignal = FHG_RIDE;
+        }
+    }
+}
+
+void EnfHG_Damage(EnfHG* this, GlobalContext* globalCtx) {
+    f32 dx;
+    f32 dz;
+    f32 dxz2;
+
+    osSyncPrintf("REVISE !!\n");
+    SkelAnime_Update(&this->skin.skelAnime);
+    Math_ApproachF(&this->warpFogR, globalCtx->lightCtx.unk_07, 1.0f, 10.0f);
+    Math_ApproachF(&this->warpFogG, globalCtx->lightCtx.unk_07, 1.0f, 10.0f);
+    Math_ApproachF(&this->warpFogB, globalCtx->lightCtx.unk_07, 1.0f, 10.0f);
+    Math_ApproachF(&this->warpFogUnk1, 0.0f, 1.0f, 5.0f);
+    Math_ApproachF(&this->actor.scale.z, 0.011499999f, 1.0f, 0.002f);
+    if (this->timers[0] != 0) {
+        Math_ApproachZeroF(&this->damageSpeedMod, 1.0f, 0.1f);
+        if (this->timers[0] == 1) {
+            this->targetPainting = this->curPainting;
+            this->inPaintingPos.x = (sPaintings[this->targetPainting].pos.x * 1.3f) + 10.0f;
+            this->inPaintingPos.y = sPaintings[this->targetPainting].pos.y;
+            this->inPaintingPos.z = (sPaintings[this->targetPainting].pos.z * 1.3f) - 3325.0f;
+        }
+    } else {
+        Math_ApproachF(&this->damageSpeedMod, 1.0f, 1.0f, 0.1f);
+    }
+    Math_ApproachF(&this->actor.world.pos.x, this->inPaintingPos.x, 1.0f, this->damageSpeedMod * this->inPaintingVelX);
+    Math_ApproachF(&this->actor.world.pos.y, 60.0f, 0.1f, 1.0f);
+    Math_ApproachF(&this->actor.world.pos.z, this->inPaintingPos.z, 1.0f, this->damageSpeedMod * this->inPaintingVelZ);
+    dx = this->actor.world.pos.x - this->inPaintingPos.x;
+    dz = this->actor.world.pos.z - this->inPaintingPos.z;
+    dxz2 = sqrtf(SQ(dx) + SQ(dz));
+    if (dxz2 < 300.0f) {
+        if (!this->spawnedWarp) {
+            this->spawnedWarp = true;
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_FHG_FIRE, this->inPaintingPos.x,
+                               this->actor.world.pos.y + 50.0f, this->inPaintingPos.z, 0,
+                               this->actor.shape.rot.y + 0x8000, 0, FHGFIRE_WARP_RETREAT);
+        }
+    }
+    if (dxz2 == 0.0f) {
+        BossGanondrof* bossFhg = BOSSFHG;
+        this->timers[0] = 140;
+        this->actionFunc = EnfHG_Retreat;
+        Animation_MorphToLoop(&this->skin.skelAnime, &gPhantomHorseAnim_00B4C8, 0.0f);
+        if (bossFhg->actor.colChkInfo.health > 24) {
+            this->bossFhgSignal = FHG_RIDE;
+        } else {
+            bossFhg->flyMode = FHG_FLY_NEUTRAL;
+        }
+        this->turnAround = -0x8000;
+    }
+}
+
+void EnfHG_Retreat(EnfHG* this, GlobalContext* globalCtx) {
+    s16 temp_rand1;
+    s16 temp_rand2;
```

Comment:
can make these camel case to fit our style. prob dont even need the word `temp_`